index_configuration_tool/tests/test_main.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import pickle
+import random
+import unittest
+from os.path import dirname
+from typing import Optional
+
+import main
+
+# Constants
+LOGSTASH_EXPECTED_OUTPUT_FILE = dirname(__file__) + "/expected_parse_output.pickle"
+TEST_KEY = "test_key"
+BASE_DATA = {
+    TEST_KEY: [("invalid_plugin1", None), ("invalid_plugin2", {})]
+}
+
+
+# Utility method to create a test plugin config
+def create_plugin_config(host_list: list[str],
+                         ssl: Optional[bool] = None,
+                         user: Optional[str] = None,
+                         password: Optional[str] = None) -> dict:
+    config = dict()
+    if len(host_list) == 1:
+        config["hosts"] = host_list[0]
+    else:
+        config["hosts"] = host_list
+    if ssl:
+        config["ssl"] = ssl
+    if user:
+        config["user"] = user
+    if password:
+        config["password"] = password
+    return config
+
+
+# Utility method to creat a test config section
+def create_config_section(plugin_config: dict) -> dict:
+    valid_plugin = (random.choice(main.SUPPORTED_ENDPOINTS), plugin_config)
+    config_section = copy.deepcopy(BASE_DATA)
+    config_section[TEST_KEY].append(valid_plugin)
+    return config_section
+
+
+class TestMain(unittest.TestCase):
+    # Run before each test
+    def setUp(self) -> None:
+        with open(LOGSTASH_EXPECTED_OUTPUT_FILE, "rb") as f:
+            self.loaded_logstash_config = pickle.load(f)
+
+    def test_get_auth(self):
+        # Empty input should produce empty auth tuple
+        self.assertFalse(main.get_auth(dict()))
```

Comment:
oof, I don't love just testing whether the result is falsey for an empty tuple. Can you do `assert main.get_auth(dict()) == tuple()` to be a little more direct about what you're looking for?

But additionally, my understanding from the `get_auth` function is that it should _actually_ be returning None, not an empty tuple (which also evaluates to False). If that's the case, can you do `assert main.get_auth(dict()) is None` and update the comment?

Reply:
Fixed the assert to be strongly-typed, as suggested. I've also broken this out into separate test cases - one for the bad inputs (where we check for `is None`) and one for the valid input